Add end-of-method mappings

tpict opened the pull request:
Currently the `]M`/`[M` behaviour is a bit wonky since it's still looking for C syntax. The mappings I've suggested will match the last non-whitespace character of a line above a class/method definition or above any line at indent level 0.

This matches the `]m`/`[m` behaviour nicely, but chokes on import statements, non-OO structures etc. I think this is acceptable as you see similar behaviour in other filetypes but I'm open to suggestions.

tpict committed:
Add end-of-method mappings

tpict committed:
Add ][, [] mappings

tpict committed:
Update last modified field

tpict commented:
The `]]`, `[[` mappings have already been defined and work well, but I've added corresponding `][`, `[]` motions.

tpict commented:
Any more thoughts on this?

sullyj3 commented:
Hey, really sorry, I completely forgot about this. Would you possibly interested in taking over as maintainer?

tpict commented:
Yeah I would! Do you have to let upstream know? I'm not sure how the transfer works

sullyj3 commented:
Sweet, go ahead and replace my name and email with yours at the top of the file, under maintainer. Then email the updated python.vim to Bram. I've let him know.

sullyj3 commented:
Let me know once you've done that, and I'll update the repo to indicate that I'm no longer the maintainer.

tpict committed:
Update maintainer credits

tpict commented:
I've just sent Bram an email. Cheers!

tpict committed:
Add .gitignore

kiryph committed:
Fix end-of-class motions in Visual mode

Stops Vim from dropping into Normal mode after an end-of-class motion

tpict committed:
Merge with upstream

tpict committed:
Fix end-of-method jumps over trailing whitespace

rohieb committed:
fix completion if compiled only with +python3

In Debian, the default vim version is not compiled with +python, but
with +python3. In this case, pythoncomplete#Complete() will throw an
error as it depends on +python. However, we can use
python3complete#Complete() if +python3 is available.

If no Python support was compiled in, pythoncomplete#Complete() will
still be used and still throw an error, so the user is warned in this
case.

Signed-off-by: Roland Hieber <r.hieber@pengutronix.de>

tpict committed:
Merge pull request #1 from rohieb/fix-python3-omnifunc

fix completion if compiled only with +python3
